lib/trino-collect/src/test/java/io/trino/collect/cache/TestEvictableCache.java

```diff
@@ -48,17 +50,75 @@
     public void testLoad()
             throws Exception
     {
-        Cache<Integer, String> cache = EvictableCache.buildWith(
-                CacheBuilder.newBuilder());
+        Cache<Integer, String> cache = EvictableCacheBuilder.newBuilder().build();
         assertEquals(cache.get(42, () -> "abc"), "abc");
     }
 
+    @Test(timeOut = TEST_TIMEOUT_MILLIS)
+    public void testEvictBySize()
+            throws Exception
+    {
+        Cache<Integer, Integer> cache = EvictableCacheBuilder.newBuilder()
+                .maximumSize(10)
+                .build();
+
+        for (int i = 0; i < 10_000; i++) {
+            int value = i * 10;
+            assertEquals((Object) cache.get(i, () -> value), value);
+        }
+        cache.cleanUp();
+        assertEquals(cache.size(), 10);
+        assertEquals(((EvictableCache<?, ?>) cache).tokensCount(), 10);
+    }
+
+    @Test(timeOut = TEST_TIMEOUT_MILLIS)
+    public void testEvictByWeight()
+            throws Exception
+    {
+        Cache<Integer, String> cache = EvictableCacheBuilder.newBuilder()
+                .maximumWeight(20)
+                .weigher((Integer key, String value) -> value.length())
+                .build();
+
+        for (int i = 0; i < 10; i++) {
+            String value = Strings.repeat("a", i);
+            assertEquals((Object) cache.get(i, () -> value), value);
+        }
+        cache.cleanUp();
+        // It's not deterministic which entries get evicted
+        int cacheSize = toIntExact(cache.size());
+        assertThat(((EvictableCache<?, ?>) cache).tokensCount()).as("tokensCount").isEqualTo(cacheSize);
+        assertThat(cache.asMap().keySet()).as("keySet").hasSize(cacheSize);
+        assertThat(cache.asMap().keySet().stream().mapToInt(i -> i).sum()).as("key sum").isLessThanOrEqualTo(20);
+        assertThat(cache.asMap().values()).as("values").hasSize(cacheSize);
+        assertThat(cache.asMap().values().stream().mapToInt(String::length).sum()).as("values length sum").isLessThanOrEqualTo(20);
+    }
+
+    @Test(timeOut = TEST_TIMEOUT_MILLIS)
+    public void testDisabledCache()
```

Comment:
I don't understand why this is `disabled`

Reply:
because it has `maximumSize(0)`